system-helper/flatpak-system-helper.c

```diff
@@ -193,6 +232,160 @@ no_progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
 {
 }
 
+static uid_t
+get_connection_uid (GDBusMethodInvocation *invocation, GError **error)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  g_autoptr(GVariant) credentials = NULL;
+  g_autoptr(GVariantIter) iter = NULL;
+  const char *key = NULL;
+  g_autoptr(GVariant) value = NULL;
+  uid_t uid;
+
+  credentials = g_dbus_connection_call_sync (connection,
+                                             "org.freedesktop.DBus",
+                                             "/org/freedesktop/DBus",
+                                             "org.freedesktop.DBus",
+                                             "GetConnectionCredentials",
+                                             g_variant_new ("(s)", sender),
+                                             NULL, G_DBUS_CALL_FLAGS_NONE, G_MAXINT, NULL, error);
+  if (credentials == NULL)
+    {
+      g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                                             "Error calling GetConnectionCredentials over D-Bus");
+      return TRUE;
+    }
+
+  g_variant_get (credentials, "(a{sv})", &iter);
+  while (g_variant_iter_loop (iter, "{&sv}", &key, &value))
+    {
+      if (strcmp (key, "UnixUserID") == 0)
+        {
+          uid = g_variant_get_uint32 (value);
+          break;
+        }
+    }
+
+  return uid;
+}
+
+static void
+mounts_changed (GUnixMountMonitor *monitor, gpointer user_data)
+{
+  FlatpakSystemHelperOngoingPull *pull = (FlatpakSystemHelperOngoingPull *) user_data;
+  g_autoptr(GList) mounts = NULL;
+  GList *l = NULL;
+
+  mounts = g_unix_mounts_get (&pull->current_timestamp);
+  for (l = g_list_reverse (mounts); l != NULL; l = l->next)
```

Comment:
Why reverse this list before scanning it?

Reply:
hmm, I was getting the mounts without the timestamp in an earlier version of the code. Therefore I used to reverse the list as the bindfs mount was getting found at the last. Less no. of loops* but anyway, with using timestamp it can be dropped. Thanks for noticing.

Reply:
Well, reversing the list requires iterating it in full..